### PR TITLE
contributor-guide.rst: explain commit message and pull request description

### PR DESCRIPTION
**Pull request by pohly**
Triggered by a discussion on IRC with @uhofemei about PR #159: the
difference between commit message and pull request description turned
out to be unclear, and our documentation was not helping.

@uhofemei: is this better?
@dbkinder: please review. This uses "::" and might conflict with PR #158.

### CODE REVIEW

File: doc/quick_start/contributor-guide.rst [hunk 1]
```diff
@@ -75,25 +75,52 @@ Here are guidelines for the review process that you and the maintainers will fol
 #. The general commit message format is::
 
      <component>: <summary>
+
      <description>
+
      [Fixes|Related-to] OP-<number>
```
Comment (gvancuts) [inline doc/quick_start/contributor-guide.rst]: While we're at it, can we add a colon in here, i.e.; [Fixes|Related-to]: OP-<number>. That would make the explanation more consistent with the example below,

Comment (pohly) [inline doc/quick_start/contributor-guide.rst]: Agreed, that would be better, because in practice we've always used the colon. Will update this.

**Comment by pohly** [comment]
And ironically, I once again forgot to add [skip ci] - too late now :-/

**Comment by dbkinder** [comment]
+1, no problems with the "::" usage.

**Comment by pohly** [comment]
Merging now to get it included in the documentation for 1.0 (albeit not exactly the released source code version).

@uhofemei comments still welcome, we should listen to feedback and can make further enhancements.

**Comment by uhofemei** [comment]
@pohly  this is clear now thanks.

**Pull request merged**